File: app/src/main/java/com/example/otb/EasyPuzzle2ViewModel.java
package com.example.otb;

import androidx.compose.runtime.MutableState;
import androidx.lifecycle.ViewModel;
import javax.inject.Inject;
import dagger.hilt.android.lifecycle.HiltViewModel;

@HiltViewModel
public class EasyPuzzle2ViewModel extends ViewModel {
    private MeasurableSensor lightSensor;
    private MutableState<Boolean> isDark;

    @Inject
    public EasyPuzzle2ViewModel(MeasurableSensor lightSensor) {
        this.lightSensor = lightSensor;
        init();
    }

    private void init() {
        lightSensor.startListening();
        lightSensor.setOnSensorValuesChangedListener(values -> {
            float lux = values.get(0);
            isDark.setValue(lux < 60f);
        });
    }

    public MutableState<Boolean> getIsDark() {
        return isDark;
    }
}

File: app/src/main/java/com/example/otb/EasyPuzzle2ViewModel.kt
package com.example.otb

import android.util.Log
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class EasyPuzzle2ViewModel @Inject constructor(
        private val lightSensor: MeasurableSensor
): ViewModel() {
    val isDark: MutableLiveData<Boolean> = MutableLiveData();
    init {
        lightSensor.startListening()
        lightSensor.setOnSensorValuesChangedListener { values ->
            val lux = values[0]
            Log.d("ARTHUR", ": $lux")
            if (lux < 10f) {
                isDark.value = true
            } else if (lux > 1000f) {
                isDark.value = false
            }
        }
    }

    override fun onCleared() {
        super.onCleared()
        lightSensor.stopListening()
    }
}